homeassistant/components/matter/__init__.py

```diff
@@ -69,29 +69,42 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     async def on_hass_stop(event: Event) -> None:
         """Handle incoming stop event from Home Assistant."""
-        await matter.disconnect()
+        await matter_client.disconnect()
 
     entry.async_on_unload(
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, on_hass_stop)
     )
 
+    # register websocket api
     async_register_api(hass)
 
-    matter.listen()
+    # launch the matter client listen task in the background
+    # use the init_ready event to keep track if it did initialize successfully
+    init_ready = asyncio.Event()
+    listen_task = asyncio.create_task(matter_client.start_listening(init_ready))
```

Comment:
What should happen if the listen task fails after config entry setup?

Reply:
that is watched above with the event and timeout or what do you mean ?

Reply:
I mean that the task can fail later, right? This task should continue in the background, right?

Reply:
ah, yes, I did not yet focus on that now. Making it more robust is for next iteration as I'm already far out of time now

Reply:
Ok.